File: src/_1ProgrammingBasicsWithJavaMay2023/_5WhileLoop/_2Exercise/_01OldBooks.java
package _1ProgrammingBasicsWithJavaMay2023._5WhileLoop._2Exercise;

import java.util.Scanner;

public class _01OldBooks {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String nameOfFavoriteBook = scanner.nextLine();
        int numberOfBooks = 0;
        boolean isFound = false;
        String findBook = scanner.nextLine();
        while (!findBook.equals("No More Books")) {
            if (findBook.equals(nameOfFavoriteBook)) {
                isFound = true;
                break;
            }
            numberOfBooks++;
            findBook = scanner.nextLine();
        }
        if (isFound == true) {
            System.out.printf("You checked %d books and found it.", numberOfBooks);
        } else {
            System.out.printf("The book you search is not here!%n");
            System.out.printf("You checked %d books.", numberOfBooks);
        }
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_5WhileLoop/_2Exercise/_01OldBooks.js
function oldBooks(input) {
    const favoriteBook = input[0];
    let currentBook = input[1];

    let count = 0;
    let found = false;

    let counter = 2;
    while (currentBook !== "No More Books") {
        if (currentBook === favoriteBook) {
            found = true;
            break;
        }
        count++;
        currentBook = input[counter++];
    }

    if (found) {
        console.log(`You checked ${count} books and found it.`);
    } else {
        console.log(`The book you search is not here!`);
        console.log(`You checked ${count} books.`);
    }

}